### DISCUSSION TITLE
How to save the rsf model in disk efficiently?

### DISCUSSION THREAD

**Discussion by ringo-alin** [kogalur/randomForestSRC#177] [Q&A] [answered]
Hi, it seems that the memory consumption of the rsf model object depends heavily on the sample size. My dataset has a size of 40,000 * 36, and the trained model (competing risk model with default parameters) uses nearly 4 Gb memory. When I save my model using `saveRDS`, it uses even more disk space (11 Gb):joy:, which takes much time to save and load. So I wonder whether there is a efficient way to save the model?

**Comment by ishwaran** [comment, marked as answer]
Yes, memory will be dependent on sample size (and other factors, such as type of forest).  To reduce the memory size of the saved object you could consider anonymous random forests using the new function `rfsrc.anonymous`.  The advantage of this method is that it does not save the training data which could potentially lead to a much smaller memory footprint.  There are some limitations with anonymous forests due to not saving the training data, so you should consult the help file to see if these limitations would be acceptable for your application.

The code would look something like:

```
o <- rfsrc.anonymous(...)
my.forest <- o$forest
save(my.forest, file = "my.forest.rda)
```

Note that you only need to save the forest for prediction on test data and not the entire grow object.

**Reply by ringo-alin** [reply]
Thank you for saving my day! 👍

**Reply by a-gialluisi** [reply]
Hi,
I tried to save my SRF using something similar to your code
_platoneSRF <- modRFSRC$forest
save(platoneSRF, file = "platone_neurodeg_final.rda")_

but then when I try to load it again, 
 _load("platone_neurodeg_final.rda")_
R reads it as a character...

Did I do something wrong?
Is there a way to save the model as an executable or even a json file, so that it can be easily implemented in informatic GUIs?

Thanks for your help.
Rgds,
Alessandro

**Reply by ishwaran** [reply]
I don't understand what the error is from your example above.

Try the following using mtcars, and see if you receive an error, and if you do, please copy and paste the error into your response

```
> o=rfsrc(mpg~.,mtcars)
> of=o$forest
> save(of, file="of.rda")
> load("of.rda")
```

For your second question, no there is no way to save it as an executable.  But the forest is a very efficient and lean data structure, especially a forest from an anonymous random forest call.  I don't think you could store it any more efficiently than it already is.

**Reply by a-gialluisi** [reply]
Great, thanks.
I got the mistake, I simply need to load it and call the model as I named it originally.
Thanks again for your help.
